var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -207,6 +214,7 @@ def url_for_version(self, version):
     conflicts('pmi=pmix', when='device=ch3')
     conflicts('pmi=pmix', when='+hydra')
     conflicts('pmi=cray', when='+hydra')
+    conflicts('vci', when='@:3.99')
```

Comment:
nitpick: can this just be `when='@:4.0'`?

Reply:
Updated with a more elegant way: by using `when='@4.0: device=ch4'` directly in the variant.

Reply:
I think you'd just want to have this variant only exist `when='@4: device=ch4'` all the way at the top. (Note that `@:4.0` means up to and including any `4.0.x` version, which is probably not what you want)

Reply:
my bad indeed, just fixed!